### ISSUE TITLE
why are the val-loss of .fit() much higher than the loss of .evaluate()?

### ISSUE THREAD

**Issue by saintthor** [keras-team/keras#12681]
I am using tensorflow.keras in https://colab.research.google.com.

tensorflow.version is 1.13.

code like:

```
for i in range( 999 ):
      mo.fit( AllTrains, [YTrain], epochs = 1, batch_size = 500, verbose=2, validation_data=( AllTests, [YTest] ))
      mo.evaluate( AllTests, [YTest] )

```
out put like:

> Train on 57591 samples, validate on 19200 samples
>  - 2s - loss: 1.0303 - val_loss: 1.1302
> 19200/19200 [==============================] - 2s 98us/sample - loss: 0.9569

the val_loss is always near 1.13 and the loss of evaluate always near 0.96.

shouldn't them be nearly same?

**Comment by saintthor** [comment]
I tried to set batch_size = 500 in evaluate, the value got higher, but still not equal to the val_loss.

and, why do I get different loss from evaluate by setting different batch_size? how to get the simplest mean loss value of the test data?

**Comment by briannemsick** [comment]
They should not be.

During each batch of training the model is updating the weights (and hopefully improving). The network weights after batch 1, 2, 3, ... n-1 are different than after the last batch of the epoch (n).

The ``loss`` from training is the average loss over the entire epoch. Meaning if the model is improving the batch losses at the beginning will be higher than at the end of the epoch.

Let's construct an example, say you have a dataset of 100 records and they are all the same. You train the network in batches of 20:
loss per train batch = [2, 1.75, 1.5, 1.25, 1] would read ``loss 1.5`` in the progress bar
validation loss  = 1  would read ``val_loss 1.0`` in the progress bar

**Comment by saintthor** [comment]
> Let's construct an example, say you have a dataset of 100 records and they are all the same. You train the network in batches of 20:
> loss per train batch = [2, 1.75, 1.5, 1.25, 1] would read `loss 1.5` in the progress bar
> validation loss = 1 would read `val_loss 1.0` in the progress bar

thank you.

for the above example, loss per train batch = [2, 1.75, 1.5, 1.25, 1] would read `loss 1.5` in the progress bar. I think it may read 1 or lower for the average value in the next epoch, right?

But in my test, the val-loss 1.13 is almost lowest. it has never reduced to 0.96.

And witch most confused me is why the loss of evaluate() may be changed by altering the batch_size. 

Dose it also train in evaluating?  Then how can I read the pure evaluation loss for the given variables to select the best?
